ext/com/concurrent_ruby/ext/SynchronizationLibrary.java

```diff
@@ -55,14 +55,14 @@ public IRubyObject initialize(ThreadContext context, IRubyObject[] args, Block b
             }
         }
 
-        @JRubyMethod(name = "synchronize", visibility = Visibility.PRIVATE)
+        @JRubyMethod(name = "synchronize", visibility = Visibility.PROTECTED)
```

Comment:
The calls you point out are both from within a module, calling synchronize against @mutex, which is presumably the IVar above. I suspect JRuby has a problem with visibility from module methods. Thanks for the info.